tests/scripts/test_generate.py

```diff
@@ -186,10 +176,12 @@ def test_unicode_characters(testdir):
         ),
     )
 
+    if (3, 6) <= sys.version_info < (3, 7):
+        testdir._pytester._monkeypatch.setenv("PYTHONIOENCODING", "utf-8")
```

Comment:
this is a private variable that is present only in recent versions of pytest. We should avoid using private APIs. Would it work if you would use the `monkeypatch` fixture instead?

```python
def test_unicode_characters(testdir, monkeypatch):
    ...
    if sys.version_info < (3, 7):
        monkeypatch.setenv("PYTHONIOENCODING", "utf-8")
```
I can't reproduce the failure though. (I checked out this PR, removed lines 179-180, run `tox -e py36-pytestlatest-coverage` and it succeeded)